src/Components/QuickSelect.tsx

```diff
@@ -1,202 +1,226 @@
-            Recently used date ranges
-          </Typography>
-          <Box mt={1} className={props.classes.flexRow}>
-            <Box className={props.classes.flexColumn}>
-              {props.recentlySelected
-                .slice(0, Math.floor(props.recentlySelected.length / 2))
-                .map((object) => (
-                  <Box>
-                    <Button
-                      onClick={() => handleClick(object)}
-                      color="primary"
-                      variant="text"
-                      size="small"
-                      disableFocusRipple={true}
-                      disableRipple={true}
-                      className={props.classes.quickSelectContainerButton}
-                    ></Button>
-                    <Box mt={-3}>
-                      <Typography color="primary" variant="subtitle2">
-                        {object.join(" ")}
-                      </Typography>
-                    </Box>
-                  </Box>
-                ))}
-            </Box>
-            <Box className={props.classes.flexColumn}>
-              {props.recentlySelected
-                .slice(Math.floor(props.recentlySelected.length / 2))
-                .map((object) => (
-                  <Box>
-                    <Button
-                      onClick={() => handleClick(object)}
-                      color="primary"
-                      variant="text"
-                      size="small"
-                      disableFocusRipple={true}
-                      disableRipple={true}
-                      className={props.classes.quickSelectContainerButton}
-                    ></Button>
-                    <Box mt={-3}>
-                      <Typography color="primary" variant="subtitle2">
-                        {object.join(" ")}
-                      </Typography>
-                    </Box>
-                  </Box>
-                ))}
-            </Box>
-          </Box>
-        </Box>
-      </Box>
-    </Box>
-  );
-};
+import React, { useContext } from "react";
+import { Divider } from "@material-ui/core";
+import { Button, Box, Typography } from "@material-ui/core";
+import { DateSelect } from "./QuickDateSelect";
+import { DateRange, TermContext } from "./DateRange";
+import { GlobalContext } from "./Constants";
+import { DateRangeUI, DropdownData } from "./Types";
+import { MSFormatter } from "./MSFormatter";
+
+interface Inputs {
+  boxClass: string;
+  recentlySelected: number[];
+  dateRange: DateRange;
+  dateRangeUI: DateRangeUI;
+  dropdownData: DropdownData;
+  setDropdownData(data: DropdownData): void;
+  setDateRangeUI(dateRangeUI: DateRangeUI): void;
+  applyChanges(dr: DateRange): void;
+  setRecentlySelected(items: number[]): void;
+}
+
+export const QuickSelect: React.FC<Inputs> = (props) => {
+  const globals = useContext(GlobalContext);
+  let out = props.dropdownData.quickSelectContent;
+
+  let commonlyUsedFirstRow = globals.commonlyUsedText.slice(
```

Comment:
Did you mean `firstColumn`, btw?

It'd be a little cleaner, too, if you pulled the length out:
```
   const listMid = Math.floor(globals.commonlyUsedText.length / 2);
   let columns = [
      globals.commonlyUsedText.slice(0, listMid),
      globals.commonlyUsedText.slice(listMid)
   ];
```

Reply:
Done